### PR TITLE
feature Add net6.0 support

### PR DESCRIPTION
**Pull request by ChrisPulman**
**What kind of change does this PR introduce?**

feature

**What is the current behaviour?**

support up to net 5

**What is the new behaviour?**

basic support for net 6

**What might this PR break?**

as Net 6 is still preview, features subject to change

**Please check if the PR fulfils these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,6 +127,14 @@ ReactiveUI is developed under an OSI-approved open source license, making it fre
 
 If you want to submit pull requests please first open a [GitHub issue](https://github.com/reactiveui/ReactiveUI/issues/new/choose) to discuss. We are first time PR contributors friendly.
 
+The current source uses **Visual Studio 2022 Preview**. Please install the following workloads to enable building the source:
```
Comment (glennawatson) [inline README.md:130]: You can run net 6 with the 2019 preview. It has to be preview though.

File: src/Directory.build.props [hunk 2]
```diff
@@ -60,11 +60,8 @@
 
   <ItemGroup>	
     <PackageReference Include="Nerdbank.GitVersioning" Version="3.4.231" PrivateAssets="all" />	
-  </ItemGroup>
 
-  <ItemGroup>
     <PackageReference Include="stylecop.analyzers" Version="1.2.0-beta.354" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" PrivateAssets="all" />
```
Comment (glennawatson) [inline src/Directory.build.props:67]: Believe we still this for the non 6 based target frameworks

File: src/ReactiveUI/ReactiveUI.csproj [hunk 3]
```diff
@@ -10,19 +10,22 @@
...
   </PropertyGroup>
 
-  <PropertyGroup Condition="$(TargetFramework) == 'net5.0-windows'">
+  <PropertyGroup Condition="$(TargetFramework) == 'net5.0-windows' or $(TargetFramework) == 'net6.0-windows'">
     <UseWpf>true</UseWpf>
     <UseWindowsForms>true</UseWindowsForms>
   </PropertyGroup>
 
-  <PropertyGroup Condition=" $(TargetFramework.StartsWith('net46')) or $(TargetFramework) == 'net47' or $(TargetFramework) == 'net471' or $(TargetFramework) == 'net5.0-windows'">
+  <PropertyGroup Condition=" $(TargetFramework.StartsWith('net46')) or $(TargetFramework) == 'net47' or $(TargetFramework) == 'net471' or $(TargetFramework) == 'net5.0-windows' or $(TargetFramework) == 'net6.0-windows'">
```
Comment (glennawatson) [inline src/ReactiveUI/ReactiveUI.csproj:18]: I think we would need separate net 6 conditionals for the different platforms like android and iOS. 

Similar to splat drawing

**Review by glennawatson — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ChrisPulman) [reply README.md]: Changed to vs2019 preview

File: src/Directory.build.props (hunk 2 above)
Comment (ChrisPulman) [reply src/Directory.build.props:67]: V6 is automatically installed with net6 sdk, a warning is given for each target if V5 is used, so will leave as V6

File: src/ReactiveUI/ReactiveUI.csproj (hunk 3 above)
Comment (ChrisPulman) [reply src/ReactiveUI/ReactiveUI.csproj]: resolved via 0fe0084

**Review by glennawatson — APPROVED** [review]

**Pull request merged**